src/dvc_objects/transfer.py

```diff
@@ -0,0 +1,39 @@
+from typing import Set
+
+from .db import ObjectDB
+
+
+def transfer(
+    src: "ObjectDB", dest: "ObjectDB", oids: Set["str"], jobs: int = None
```

Comment:
There's a question of how `oids` will be discovered by the caller. I guess it's a responsibility of `odb`.

See https://github.com/iterative/dvc/pull/7874.